kubectl-volsync/cmd/migration_create.go

```diff
@@ -72,34 +108,294 @@ func validateMigrationCreate(cmd *cobra.Command, args []string) error {
+		return nil, err
+	}
+	mc.mr.data.Destination = mrd
+
+	k8sClient, err := newClient(mrd.Cluster)
+	if err != nil {
+		return nil, err
+	}
+	mc.clientObject = k8sClient
+
+	// build struct migrationRelationshipDestination from cmd line args
+	err = newMigrationRelationshipDestination(mc, cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	return mc, nil
+}
+
+func parsePVCName(cmd *cobra.Command, mrd *migrationRelationshipDestination) error {
+	pvcname, err := cmd.Flags().GetString("pvcname")
+	if err != nil {
+		return fmt.Errorf("failed to fetch the pvcname, err = %w", err)
+	}
+	xcr, err := ParseXClusterName(pvcname)
+	if err != nil {
+		return fmt.Errorf("failed to parse cluster name from pvcname, err = %w", err)
 	}
-	rName, err := cmd.Flags().GetString("relationship")
+	mrd.PVCName = xcr.Name
+	mrd.Namespace = xcr.Namespace
+	mrd.Cluster = xcr.Cluster
+
+	return nil
+}
+
+//nolint:funlen
+func newMigrationRelationshipDestination(mc *migrationCreate, cmd *cobra.Command) error {
+	mrd := mc.mr.data.Destination
+
+	cm, err := cmd.Flags().GetString("copymethod")
+	if err != nil {
+		return fmt.Errorf("failed to fetch the copy method, err = %w", err)
+	}
+	mrd.Destination.CopyMethod = volsyncv1alpha1.CopyMethodType(cm)
+	if mrd.Destination.CopyMethod != volsyncv1alpha1.CopyMethodDirect &&
+		mrd.Destination.CopyMethod != volsyncv1alpha1.CopyMethodClone &&
+		mrd.Destination.CopyMethod != volsyncv1alpha1.CopyMethodSnapshot {
+		return fmt.Errorf("unsupported copymethod: %v", mrd.Destination.CopyMethod)
+	}
+
+	mc.PVC = mc.getDestinationPVC(cmd.Context())
+	if mc.PVC == nil {
+		cap, err := cmd.Flags().GetString("capacity")
+		if err != nil || cap == "" {
+			return fmt.Errorf("please provide the capacity or create a PVC by name: %s", mrd.PVCName)
+		}
+
+		capacity, _ := resource.ParseQuantity(cap)
+		mrd.Destination.Capacity = &capacity
+
+		accessMode, err := cmd.Flags().GetString("accessmodes")
+		if err != nil {
+			return fmt.Errorf("failed to fetch access mode, %w", err)
+		}
+
+		if v1.PersistentVolumeAccessMode(accessMode) != v1.ReadWriteOnce &&
+			v1.PersistentVolumeAccessMode(accessMode) != v1.ReadOnlyMany &&
+			v1.PersistentVolumeAccessMode(accessMode) != v1.ReadWriteMany &&
+			v1.PersistentVolumeAccessMode(accessMode) != v1.ReadWriteOncePod {
+			return fmt.Errorf("unsupported access mode: %v", accessMode)
+		}
+		accessModes := []v1.PersistentVolumeAccessMode{v1.PersistentVolumeAccessMode(accessMode)}
+		mrd.Destination.AccessModes = accessModes
+
+		storageClass, err := cmd.Flags().GetString("storageclass")
+		if err != nil || storageClass == "" {
+			klog.Infof("storage class not provided, binding to default storage class")
+		}
+		mrd.Destination.StorageClassName = &storageClass
+	} else {
+		mrd.Destination.Capacity = mc.PVC.Spec.Resources.Requests.Storage()
+		mrd.Destination.AccessModes = mc.PVC.Spec.AccessModes
+		mrd.Destination.StorageClassName = mc.PVC.Spec.StorageClassName
+		mrd.Cluster = mc.PVC.ClusterName
+	}
```

Comment:
> to clarify, the capacity, accessmodes, and storageclass should only be set in the ReplicationDestination if DestinationPVC is not set

This is in the case of pre-existing PVC right? Are we not updating the relationship file with capacity, accessmodes and storageclass if PVC already present ?

Reply:
If DestinationPVC is set, the operator will not create a PVC (it'll just use the DestinationPVC one). Capacity, accessmodes, and storageclass (in .spec.rsync) are only used by the operator when creating a PVC.

case 2 is important because if the operator creates the PVC, the PVC will be deleted when the ReplicationDestination object is deleted, and we don't want that to happen if we're using Direct/None.

Reply:
Handling copymethod=Direct only for the migration use case. So considering case 1 and case 2  as per our discussion